config/routes.rb

```diff
@@ -1,10 +1,9 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Reveal health status on /up that returns 200 if the app boots with no exceptions, otherwise 500.
-  # Can be used by load balancers and uptime monitors to verify that the app is live.
   get "up" => "rails/health#show", as: :rails_health_check
 
-  # Defines the root path route ("/")
-  # root "posts#index"
+  get 'users', to: 'users#index', as: :users  
+  get 'users/:id', to: 'users#show', as: :user 
+
+  get 'users/:user_id/posts', to: 'posts#index', as: :user_posts  
+  get 'posts/:id', to: 'posts#show', as: :user_post               
```

Comment:
- [OPTIONAL] While this is going to work fine, I will encourage you to take advantage of resources by having something like:
```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

With this, you have shorter code to write and you keep it simple, stupid.